crates/ide/src/display/navigation_target.rs

```diff
@@ -376,6 +377,23 @@ impl ToNav for hir::TypeParam {
     }
 }
 
+impl ToNav for hir::LifetimeParam {
+    fn to_nav(&self, db: &RootDatabase) -> NavigationTarget {
+        let src = self.source(db);
+        let full_range = src.value.syntax().text_range();
+        NavigationTarget {
+            file_id: src.file_id.original_file(db),
+            name: self.name(db).to_string().into(),
+            kind: LIFETIME_PARAM,
```

Comment:
Unrelated to PR, but, now that I look at it, using `SyntaxKind` here is abstraction sloppiness. This really should be `enum DefinitionKind` declated in the `ide` crate, and re-used by `HighlightTag` enum. Will probably look into fixing that.